docs/src/main/tut/json.md

```diff
@@ -4,183 +4,253 @@ weight: 310
+User("Alice").asJson
 ```
 
-And clean everything up.
+Equipped with an `Encoder` and `.asJson`, we can send JSON in requests
+and responses for our case classes:
 
 ```tut:book
-httpClient.shutdownNow()
-blazeServer.shutdownNow()
+Ok(Hello("Alice").asJson).run
+POST(uri("/hello"), User("Bob").asJson).run
 ```
 
-## Talking to the Github API
-
-As an example, we'll use the [github-orgs] endpoint. The data is shortened a bit
-for clarity.
-
-```json
-[
-  {
-    "id": 3692188,
-    "name": "http4s",
-    "full_name": "http4s/http4s",
-    "owner": {
-      "login": "http4s",
-      "id": 1527492
-      // ...
-    },
-    "private": false,
-    "html_url": "https://github.com/http4s/http4s",
-    "description": "A minimal, idiomatic Scala interface for HTTP",
-    "size": 21365,
-    "stargazers_count": 533,
-    "language": "Scala",
-    "forks_count": 118,
-    "open_issues_count": 69,
-    "forks": 118,
-    "open_issues": 69,
-    "watchers": 533
-    // ...
-  }
-  // ...
-]
-```
+## Receiving raw JSON
+
+Just as we needed an `EntityEncoder[JSON]` to send JSON from a server
+or client, we need an `EntityDecoder[JSON]` to receive it.
 
-To capture this structure, we'll need corresponding case classes.
+The `org.http4s.circe._` package provides an implicit
+`EntityDecoder[Json]`.  This makes it very easy to decode a request or
+response body to JSON using the `as` syntax:
 
 ```tut:book
-case class User(login: String, id: Long)
-case class Repo(id: Long, name: String, full_name: String, owner: User, `private`: Boolean, html_url: String, description: String, size: Int, stargazers_count: Int, language: String, forks_count: Int, open_issues_count: Int, forks: Int, open_issues: Int, watchers: Int)
+Ok("""{"name":"Alice"}""").as[Json].run
+POST(uri("/hello"),"""{"name":"Bob"}""").as[Json].run
 ```
 
-This parts skips over the [client] explanation. We'll use circe, with
-[circe-generic] for codec derivation. The JSON decoder is provided by
-circe-generic, and passed to the `client.expect` method via the second
-argument, which is usually an implicit `EntityDecoder`.  In this case,
-we want to be more explicit.
+Like sending raw JSON, this is useful to a point, but we typically
+want to get to a typed model as quickly as we can.
+
+## Decoding JSON to a case class
 
-<!-- For more information about the uri templating, visit [uri]. -->
+To get from an HTTP entity to `Json`, we use an `EntityDecoder[Json]`.
+To get from `Json` to any type `A`, we need an `io.circe.Decoder[A]`.
+http4s-circe provides the `jsonOf` function to make the connection all
+the way from HTTP to your type `A`.  Specifically, `jsonOf[A]` takes
+an implicit `Decoder[A]` and makes a `EntityDecoder[A]`:
 
 ```tut:book
-import scalaz.concurrent.Task
+Ok("""{"name":"Alice"}""").as(jsonOf[User]).run
+POST(uri("/hello"), """{"name":"Bob"}""").as(jsonOf[User]).run
+```
 
-import org.http4s.util.string._
+Note the argument to `as` is in parentheses instead of square
```

Comment:
I can follow this, but it's a bit confusing as it's behaving at both a type level and at the function level. However as it's high level documentation I don't think it's detrimental if they can follow it. We may want to allude to where information on this can be found. As finding the word "as" in a code base is close to impossible.